File: frontend/src/types/image.ts
// 消息类型
export type MessageType = 'text' | 'image';

// 对话消息
export interface Message {
  role: string; // 角色: "user" | "assistant"
  type: MessageType; // 消息类型: "text" | "image"
  content?: string; // 文本内容 (type="text" 时有效)
  url?: string; // 图片 URL (type="image" 时有效)
}

// 图片生成请求
export interface ImageGenerateRequest {
  prompt: string;
  images?: string[]; // OSS 中的图片路径列表
  workspace?: string; // 工作区名称（可选，用于生成图片存储）
  messages?: Message[]; // 完整的对话历史 (可选，用于记录)
  enable_web_search?: boolean; // 是否启用联网搜索（默认 false）
}

// 图片上传响应
export interface ImageUploadResponse {
  path: string; // OSS 中的图片路径
  url: string; // 图片访问 URL
}

// 生成的内容片段
export interface GeneratePart {
  type: 'text' | 'image';
  text?: string;
  image?: GeneratedImage;
}

// 生成的图片
export interface GeneratedImage {
  data?: string; // Base64 编码（可选，用于兼容）
  mimeType: string;
  path?: string; // OSS 中的图片路径
  url?: string; // 图片访问 URL
}

// 图片生成响应
export interface ImageGenerateResponse {
  parts: GeneratePart[];
}

// 图片信息
export interface ImageInfo {
  id: number; // 图片 ID
  path: string; // OSS 中的路径
  url: string; // 图片访问 URL（原图）
  thumbnail_url: string; // 缩略图访问 URL
  name: string; // 文件名
  size: number; // 文件大小（字节）
  updated: string; // 最后更新时间（ISO 8601 格式）
  source_type: 'upload' | 'generate'; // 来源类型
  prompt?: string; // 生成时的提示词
  ref_images?: string[]; // 生成时的引用图片
  message_list?: Message[]; // 生成时的对话历史
}

// 列出工作区图片响应
export interface ListWorkspaceImagesResponse {
  images: ImageInfo[];
}

// 删除图片请求
export interface DeleteImageRequest {
  path: string; // OSS 中的图片路径
}

// 重命名图片请求
export interface RenameImageRequest {
  path: string; // OSS 中的图片路径
  new_name: string; // 新文件名
  workspace: string; // 工作区名称
}

// 重命名图片响应
export interface RenameImageResponse {
  image: ImageInfo; // 重命名后的图片信息
}

// 获取图片详情响应
export interface GetImageDetailResponse {
  image: ImageInfo; // 图片详细信息（包含 message_list）
}

// API 响应包装
export interface ApiResponse<T> {
  code: number;
  message: string;
  data?: T;
}

File: backend/internal/model/image.go
package model


// ImageGenerateResponse 图片生成响应
type ImageGenerateResponse struct {
	Parts []GeneratePart `json:"parts"` // 有序的内容片段
}

// GeneratePart 生成内容片段
type GeneratePart struct {
	Type  string          `json:"type"`            // 类型: "text" | "image"
	Text  string          `json:"text,omitempty"`  // 文本内容 (Type="text" 时有效)
	Image *GeneratedImage `json:"image,omitempty"` // 图片信息 (Type="image" 时有效)
}

// GeneratedImage 生成的图片信息
type GeneratedImage struct {
	Data     string `json:"data,omitempty"`     // Base64 编码的图片数据（可选，用于兼容）
	MimeType string `json:"mimeType"`           // 图片 MIME 类型
	Path     string `json:"path,omitempty"`     // OSS 中的图片路径
	URL      string `json:"url,omitempty"`      // 图片访问 URL
}

// ImageUploadRequest 图片上传请求
type ImageUploadRequest struct {
	Workspace string `form:"workspace" binding:"required"` // 工作区名称
}

// ImageUploadResponse 图片上传响应
type ImageUploadResponse struct {
	Path string `json:"path"` // OSS 中的图片路径
	URL  string `json:"url"`  // 图片访问 URL
}

// ImageGenerateRequest 图片生成请求
type ImageGenerateRequest struct {
	Prompt    string   `json:"prompt" binding:"required"`
	Images    []string `json:"images"`              // OSS 中的图片路径列表
	Workspace string   `json:"workspace"`           // 工作区名称（可选，用于生成图片存储）
}

// ListWorkspaceImagesRequest 列出工作区图片请求
type ListWorkspaceImagesRequest struct {
	Workspace string `form:"workspace" binding:"required"` // 工作区名称
}

// ImageInfo 图片信息
type ImageInfo struct {
	Path         string `json:"path"`          // OSS 中的路径
	URL          string `json:"url"`           // 图片访问 URL（原图）
	ThumbnailURL string `json:"thumbnail_url"` // 缩略图访问 URL
	Name         string `json:"name"`          // 文件名
	Size         int64  `json:"size"`         // 文件大小（字节）
	Updated      string `json:"updated"`      // 最后更新时间（ISO 8601 格式）
}

// ListWorkspaceImagesResponse 列出工作区图片响应
type ListWorkspaceImagesResponse struct {
	Images []ImageInfo `json:"images"` // 图片列表
}

// DeleteImageRequest 删除图片请求
type DeleteImageRequest struct {
	Path string `json:"path" binding:"required"` // OSS 中的图片路径
}

// RenameImageRequest 重命名图片请求
type RenameImageRequest struct {
	Path      string `json:"path" binding:"required"`      // OSS 中的图片路径
	NewName   string `json:"new_name" binding:"required"`  // 新文件名
	Workspace string `json:"workspace" binding:"required"`  // 工作区名称
}

// RenameImageResponse 重命名图片响应
type RenameImageResponse struct {
	Image ImageInfo `json:"image"` // 重命名后的图片信息
}
